java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -0,0 +1,274 @@
+        TextView tv = (TextView) recordNameLayout.findViewById(R.id.current_record_name);
+
+        HealthVaultApp application = HealthVaultApp.getInstance();
+		if(mService.isAppConnected() && application.getCurrentRecord() != null) {
+	       tv.setText(application.getCurrentRecord().getName());
+
+	       recordNameLayout.setOnClickListener(new OnClickListener() {
+
+				@Override
+				public void onClick(View arg0) {
+					Intent intent = new Intent(MainActivity.this, RecordPickerActivity.class);
+					startActivity(intent);
+				}
+			});
+		}
+		else {
+			tv.setVisibility(View.GONE);
+		}
+
+		return true;
+	}
+
+	public void onConnected() {
+	}
+
+	public void onError(Exception e) {
+		Toast.makeText(
+                MainActivity.this, 
+                e.getMessage(), 
+                Toast.LENGTH_LONG).show();
+	}
+
+	@Override
+	public boolean onOptionsItemSelected(MenuItem item) {
+	    switch (item.getItemId()) {
+	        case R.id.action_connect:
+	            doConnect();
+	            return true;
+	        default:
+	            return super.onOptionsItemSelected(item);
+	    }
+	}
+	
+	private void doConnect() {
+		if (!mService.isAppConnected()) {
+			connectProgressDialog = ProgressDialog.show(
+                    MainActivity.this,
+                    "",
+                    "Please wait...", 
+                    true);
+			
+			HealthVaultSettings settings = mService.getSettings();
+			settings.setMasterAppId("c6ba979f-c342-4408-a2bc-0dfb43b2bf8d");
+			settings.setServiceUrl("https://platform.healthvault-ppe.com/platform/wildcat.ashx");
+			settings.setShellUrl("https://account.healthvault-ppe.com");
+			settings.setIsMultiInstanceAware(true);
+			settings.setIsMRA(true);
+			mService.start(MainActivity.this, MainActivity.this);
+    	}
+		else {
+			Toast.makeText(MainActivity.this, "App is already connected", Toast.LENGTH_LONG).show();
+		}
+	}
+	
+	public void onListItemClick(ListView parent, View v, int position, long id) {
+        TextView item = (TextView) v;
+        Intent intent = null;
+        if (mService.isAppConnected()) {
+	        switch(position) {
+	        case 0: 
+        		intent = new Intent(MainActivity.this, WeightActivity.class);
+	        }
+        }
+        
+        if(intent != null) {
+        	startActivity(intent);
+        }
+    }
+	
+	private String writeFile() {
+		String filename = "writefile" + (int)(Math.random() * 100)  + ".txt";
+		String string = "This is from file upload";
+		FileOutputStream outputStream = null;
+
+		try {
+		  outputStream = openFileOutput(filename, Context.MODE_PRIVATE);
+		  outputStream.write(string.getBytes());
+
+		  return filename;
+		} catch (Exception e) {
+		  e.printStackTrace();
+		} finally {
+			try {
+				outputStream.close();
+			} catch (Exception e){
+			}
+		}
+
+		return null;
+	}
+	
+	private class HVConnect extends AsyncTask<Void, Void, Void> {
+		
+		private ProgressDialog progressDialog;
```

Comment:
nit: Seems like more indentation issues throughout. Should set IDE settings to either always use tabs or spaces for tabs, but not a mix of both. Consider using the code formatting abilities cooked into android studio